webhooks/validating/configuration/src/main.rs

```diff
@@ -0,0 +1,765 @@
+        }
+        serde_json::Value::Array(s) => {
+            for (pos, _e) in s.iter().enumerate() {
+                if let Err(e) = check(&s[pos], &deserialized[pos]) {
+                    return Err(None.ok_or(format!(
+                        "input index ({:?}) not equal to parsed: ({:?})",
+                        pos, e
+                    ))?);
+                }
+            }
+            Ok(())
+        }
+        serde_json::Value::String(s) => match deserialized {
+            serde_json::Value::String(ds) => {
+                if s != ds {
+                    Err(None.ok_or(format!("input ({:?}) not equal to parsed ({:?})", s, ds))?)
+                } else {
+                    Ok(())
+                }
+            }
+            _ => Err(None.ok_or(format!(
+                "input ({:?}) not equal to parsed ({:?})",
+                s, deserialized
+            ))?),
+        },
+        serde_json::Value::Bool(b) => match deserialized {
+            serde_json::Value::Bool(db) => {
+                if b != db {
+                    Err(None.ok_or(format!("input ({:?}) not equal to parsed ({:?})", b, db))?)
+                } else {
+                    Ok(())
+                }
+            }
+            _ => Err(None.ok_or(format!(
+                "input ({:?}) not equal to parsed ({:?})",
+                b, deserialized
+            ))?),
+        },
+        serde_json::Value::Number(n) => match deserialized {
+            serde_json::Value::Number(dn) => {
+                if n != dn {
+                    Err(None.ok_or(format!("input ({:?}) not equal to parsed ({:?})", n, dn))?)
+                } else {
+                    Ok(())
+                }
+            }
+            _ => Err(None.ok_or(format!(
+                "input ({:?}) not equal to parsed ({:?})",
+                n, deserialized
+            ))?),
+        },
+        serde_json::Value::Null => match deserialized {
+            serde_json::Value::Null => Ok(()),
+            _ => Err(None.ok_or(format!(
+                "input (Null) not equal to parsed ({:?})",
+                deserialized
+            ))?),
+        },
+    }
+}
+
+fn filter_configuration(mut v: Value) -> Value {
+    let metadata = v["metadata"].as_object_mut().expect("Object");
+    metadata.remove("creationTimestamp");
+    metadata.remove("deletionTimestamp");
+    metadata.remove("managedFields");
+
+    let generation = metadata.get_mut("generation").expect(".generation");
+    *generation = json!(generation.as_f64().expect("integer"));
+
+    v
+}
+fn validate_configuration(rqst: &AdmissionRequest) -> AdmissionResponse {
+    println!("Validating Configuration");
+    match &rqst.object {
+        Some(raw) => {
+            let x: RawExtension = serde_json::from_value(raw.clone()).expect("RawExtension");
+            let y = serde_json::to_string(&x).expect("success");
+            let c: KubeAkriConfig = serde_json::from_str(y.as_str()).expect("success");
```

Comment:
Possibly make the `expect` message[s] more descriptive, like "could not parse as an Akri Configuration" for this one, or just use unwrap and have the `serde_json` error message be displayed?

Reply:
Fixed: https://github.com/deislabs/akri/pull/206/commits/6fc791eab3ea8e924033d6233a89198194ee4d70